File: Shantanu_Mamgain/Two-Pointer-Leetcode/Container-With-Most-Water.java
class Solution {
    public int maxArea(int[] height) {
        int j = height.length-1;
        int i = 0;
        int count = 0;

        while(i <= j){
            count = Math.max(count , Math.min(height[i], height[j]) * (j-i));
            if(height[i] < height[j]){
                i++;
            }
            else{
                j--;
            }

        }

        return count;
    }
}

File: Lakshay Garg/Leetcode/Two_Pointers_Leetcode/Container_With_Most_Water.cpp
class Solution {
    public int maxArea(int[] h) {
        int n=h.length;
        int i=0,j=n-1,ans=0;
        int count =0;
        while(i<=j){
            count=Math.min(h[i],h[j])*(j-i);
            ans=Math.max(ans,count);
            if(h[i]>h[j]) j--;
            else i++;
        }
        return ans;
    }
}
